bigquery/google/cloud/bigquery/table.py

```diff
@@ -783,43 +784,72 @@ def __init__(self, values, field_to_index):
         self._xxx_field_to_index = field_to_index
 
     def values(self):
-        return self._xxx_values
+        """Return the values included in this row.
 
-    def keys(self):
+        Returns:
+            Sequence[object]: A sequence of length ``len(row)``.
         """
-        Return keys as of a dict:
-        >>> Row(('a', 'b'), {'x': 0, 'y': 1}).keys()
-        ['x', 'y']
+        return copy.deepcopy(self._xxx_values)
```

Comment:
@tswast Sorry, I wasn't trying to suggest you actually do a copy. The constructor "mandates" a `tuple` and it certainly would be more performant to just return the immutable `values`

Reply:
I've just updated `values()` to return an iterator/generator to be more consistent with [dict.values()](https://docs.python.org/3.6/library/stdtypes.html#dict.values). I'll keep the copy for the reason that the items might be mutable objects.